components/src/components/event-list/event-list.tsx

```diff
@@ -0,0 +1,193 @@
+    @Prop({ mutable: true })
+    events: string;
+
+    // The transformed webinar and event data ready to
+    // turned into listings.
+    @State()
+    pulumiEvents: PulumiEvent[] = [];
+
+    // The current choice of the event filter.
+    @State()
+    eventFilterChoice: EventFilter = "all";
+
+    componentWillLoad() {
+        // Parse the data strings.
+        const webinars = parseWebinarsString(this.webinars);
+        const events = parseEventsString(this.events).filter((event) => new Date().getTime() <= event.startDate.getTime());
+
+        // Set the event data.
+        this.pulumiEvents = this.pulumiEvents.concat(webinars).concat(events);
+    }
+
+    // Set the item filter when a filter item is clicked.
+    private handleEventFilterItemClick(filter: EventFilter) {
+        this.eventFilterChoice = filter;
+    }
+
+    private renderEventFilterItem(args: EventFilterItem) {
+        // Mark an item as active if it is selected.
+        const itemClass = classNames("w-1/12 text-center mx-5 event-filter-item", {
+            "active": this.eventFilterChoice === args.key,
+        });
+
+        return(
+            <li class={itemClass} onClick={() => this.handleEventFilterItemClick(args.key)}>
+                <i class={`fas fa-${args.icon} text-4xl`}></i>
+                <p class="hidden sm:block m-0 mt-3">{args.text}</p>
+            </li>
+        );
+    }
+
+    private renderEventFilter() {
+        // The event filter data.
+        const items: EventFilterItem[] = [
+            { text: "All", icon: "asterisk", key: "all" },
+            { text: "Featured", icon: "star", key: "featured" },
+            { text: "Upcoming Sessions", icon: "users", key: "upcoming" },
+            { text: "Videos", icon: "video", key: "on-demand" },
+            { text: "PulumiTV", icon: "tv", key: "pulumitv" },
+        ];
+
+        // Get the selected choice.
+        const selectedChoice = items.find((item) => item.key === this.eventFilterChoice);
+
+        return(
+            <div class="w-full mb-5">
+                <ul class="flex list-none event-list-filter">
+                    { items.map(this.renderEventFilterItem.bind(this)) }
+                </ul>
+                <h2 class="px-5 mt-0">{selectedChoice.text}</h2>
+            </div>
+        );
+    }
+
+    private renderEventListRow(events: PulumiEvent[]) {
+        return(
+            <ul class="flex flex-wrap justify-center list-none p-0">
+                {events.map((event, i) => {
+                    return(
+                        <pulumi-event-list-item key={i} event={event}></pulumi-event-list-item>
+                    );
+                })}
+            </ul>
+        );
+    }
+
+    render() {
+        // Filter the events based on the currently selected filter.
+        const filteredEvents = this.eventFilterChoice === "all" ? this.pulumiEvents : this.pulumiEvents.filter((event) => {
+            return event.type.indexOf(this.eventFilterChoice) > -1;
```

Comment:
I think if `type` were renamed to the plural (since it looks like it could be multiple types?), this could be simplified a bit as `event.types.includes(this.eventFilterChoice)`. Perhaps more, but it took me a bit to figure out what was happening here based on the naming.